src/MIDebugEngine/Engine.Impl/DebuggedProcess.cs

```diff
@@ -1422,29 +1422,53 @@ public ReadOnlyCollection<RegisterGroup> GetRegisterGroups()
             await _breakpointManager.EnableAfterFuncEvalAsync();
         }
 
+        /// <summary>
+        /// Finds the line associated with a start address.
+        /// </summary>
+        public async Task<uint> LineForStartAddress(string file, ulong startAddress)
+        {
+            List<ulong> addresses = new List<ulong>();
+            SourceLineMap srcLines = await SourceLineCache.GetLinesForFile(file);
+            if (srcLines == null || srcLines.Count == 0)
+            {
+                srcLines = await SourceLineCache.GetLinesForFile(System.IO.Path.GetFileName(file));
+            }
+            if (srcLines == null || srcLines.Count == 0)
+            {
+                return 0;
+            }
+
+            SourceLine srcLine;
+            if (srcLines.TryGetValue(startAddress, out srcLine))
+            {
+                return srcLine.Line;
+            }
+            return 0;
+        }
+
         public async Task<List<ulong>> StartAddressesForLine(string file, uint line)
         {
             List<ulong> addresses = new List<ulong>();
             var srcLines = await SourceLineCache.GetLinesForFile(file);
```

Comment:
I agree. I was trying to not use it in the code I wrote, but left it in existing code. I can be more aggressive about changing it in related code.
I still use var in foreach's statements though, I think I've found a few cases in foreach when it implicitly tries to cast to the type specified resulting in a runtime error instead of a compile error (I think they added this in the .NET 1.0 days to make working with collection easier).